tests/src/test/scala/org/http4s/StaticFileSpec.scala

```diff
@@ -283,7 +284,27 @@ class StaticFileSpec extends Http4sSpec with Http4sLegacyMatchersIO {
         .fromURL[IO](getClass.getResource("/foo"), testBlocker)
         .value
         .unsafeRunSync
-      s must_== None
+      s must beNone
+    }
+
+    "return none from a URL that points to a resource that does not exist" in {
+      val s = StaticFile
+        .fromURL[IO](new URL("https://github.com/http4s/http4s/fooz"), testBlocker)
+        .value
+        .unsafeRunSync
+      s must beNone
+    }
+
+    "raise exception when url does not exist" in {
```

Comment:
This case is a really good catch! The only thing I could mildly recommend is to try leveraging a `throwA` matcher that should make the following check as simple as
```
StaticFile
          .fromURL[IO](new URL("https://quuzgithubfoo.com/http4s/http4s/fooz"), testBlocker)
          .value
          .unsafeRunSync must throwA[UnknownHostException]
```
https://etorreborre.github.io/specs2/guide/SPECS2-4.10.0/org.specs2.guide.Matchers.html#out-of-the-box

Reply:
makes sense